pandevice/base.py

```diff
@@ -2775,10 +2781,10 @@ class XapiWrapper(pan.xapi.PanXapi):
         def __init__(self, *args, **kwargs):
             self.pan_device = kwargs.pop('pan_device', None)
             pan.xapi.PanXapi.__init__(self, *args, **kwargs)
-
+            pred = lambda x: inspect.ismethod(x) or inspect.isfunction(x)
```

Comment:
Python3's inspect module has slightly different behavior with ismethod in python2. inspect.ismethod returns true for unbound and bound methods. In Python3 there is no notion of unbound methods so ismethod returns only bound methods. Perhaps the pan-python methods were unbound causing it to fail with Python3?
"Unbound methods" are functions according to Python3, but they are methods in python2.
Thus, I needed the pred function to get the tests to work with Python2 and Python3.
TL;DR:
inspect.ismethod works with Python2 but not Python3
inspect.isfunction works with Python3 but not Python2